pkg/test/case.go

```diff
@@ -544,13 +545,24 @@ func newClient(kubeconfig string) func(bool) (client.Client, error) {
 	}
 }
 
-func newDiscoveryClient(kubeconfig string) func() (discovery.DiscoveryInterface, error) {
+func newDiscoveryClient(kubeconfig, context string) func() (discovery.DiscoveryInterface, error) {
 	return func() (discovery.DiscoveryInterface, error) {
-		config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+		config, err := buildConfigWithContext(kubeconfig, context)
 		if err != nil {
 			return nil, err
 		}
 
 		return discovery.NewDiscoveryClientForConfig(config)
 	}
 }
+
+func buildConfigWithContext(kubeconfig, context string) (*rest.Config, error) {
```

Comment:
Can you please put this function in `pkg/k8s`? I have a hunch there might be a request for supporting `--context` command line flag, and then it would be good to have it handy there.